src/main/webapp/WEB-INF/view/activity.jsp

```diff
@@ -67,21 +62,32 @@
       %>
       <hr/>
       <strong> Users: </strong>
-      <ul class="mdl-list">
       <%
-        for(User user : users){
+      if(users == null || users.isEmpty()){
+      %>
+        <p>Aww nothing so far!</p>
+      <%
+      }
+      else{
+      %>
+        <ul class="mdl-list">
+      <%
+        for(int i=users.size()-1; i>=0; i--){
```

Comment:
Similar suggestion here about sorting in the servlet instead of making an ordering assumption here.